src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs

```diff
@@ -1099,24 +1127,25 @@ private void InitializeFSM()
                     }
                     awh.HandlerListener.ContinueWith(result => result.Result.Notify(disassociateNotification),
                         TaskContinuationOptions.ExecuteSynchronously);
+                    awh.WrappedHandle.Disassociate(DisassociationReason(@event.Reason), _log);
```

Comment:
This may have been the source of the leak - failed to disassociate handles when handshake failed at the finish line.